srcCpp/ParameterManager.cxx

```diff
@@ -0,0 +1,945 @@
+                output[static_cast<GROUP>(i)] += get_center_header_help_line("SECURE");
+                break;
+          #endif
+            case RAWTRANSPORT:
+                output[static_cast<GROUP>(i)] += get_center_header_help_line("RAWTRANSPORT");
+                break;
+            default:
+                break;
+        }
+    }
+
+    oss << "/**********************************************************************************************/\n"
+        << "Usage:\t perftest_cpp [options]\n"
+        << "Where [options] are:\n";
+    output[GENERAL] +="\t-help                           - Print this usage message and exit\n";
+
+    for (it = parameterList.begin(); it != parameterList.end(); it++) {
+        if (!it->second.get()->get_internal()) {
+            output[it->second.get()->get_group()] +=
+                    it->second.get()->print_command_line_parameter();
+        }
+    }
+    std::map<GROUP, std::string>::iterator itOutput;
+    for (itOutput = output.begin(); itOutput != output.end(); itOutput++) {
+        oss << itOutput->second;
+    }
+    oss << "/**********************************************************************************************/\n";
+    return oss.str();
+}
+
+// check -help option
+bool ParameterManager::check_help(int argc, char *argv[])
+{
+    std::vector<std::string> allArgs(argv, argv + argc);
+    for (unsigned int i = 1; i < allArgs.size(); i++) {
+        if (allArgs[i] == "-help" || allArgs[i] == "-h") {
+            std::cout << display_help() <<'\n';
+            return true;
+        }
+    }
+    return false;
+}
+
+ParameterManager::~ParameterManager()
+{
+    parameterList.clear();
+}
+
+// check if a variable has been set
+bool ParameterManager::is_set(std::string parameterKey)
+{
+    std::map<std::string, AnyParameter>::iterator it;
+    it = parameterList.find(parameterKey);
+    if (it != parameterList.end()) {
+        return parameterList[parameterKey].get()->get_isSet();
+    } else {
+        return false;
+    }
+}
+
+
+bool ParameterManager::validate_group()
+{
+    bool success = true;
+    std::map<std::string, AnyParameter>::iterator it;
+    for (it = parameterList.begin(); it != parameterList.end(); it++) {
+        if (it->second.get()->get_isSet()) {
+            if (it->second.get()->get_group() == PUB && GetInstance().get<bool>("sub")) {
+                fprintf(stderr, "Cannot use '%s' while setting '-sub'.\n",
+                        it->second.get()->get_command_line_argument().first.c_str());
+                success = false;
+            } else if (it->second.get()->get_group() == SUB && GetInstance().get<bool>("pub")) {
+                fprintf(stderr, "Cannot use '%s' while setting '-pub'.\n",
+                        it->second.get()->get_command_line_argument().first.c_str());
+                success = false;
+            }
+        }
+    }
+    return success;
+}
+
+
+std::vector<std::string> ParameterManager::split(std::string str, char delimiter)
```

Comment:
You can use here the [strtok](http://www.cplusplus.com/reference/cstring/strtok/) function.

Reply:
Done